test/storage.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+function storageMock() {
+    let storage = {};
+
+    return {
+        setItem: function(key, value) {
+            storage[key] = value || '';
+        },
+        getItem: function(key) {
+            return key in storage ? storage[key] : null;
+        },
+        removeItem: function(key) {
+            delete storage[key];
+        },
+        get length() {
+            return Object.keys(storage).length;
+        },
+        key: function(i) {
+            let keys = Object.keys(storage);
+            return keys[i] || null;
+        }
+    };
+}
+
+// mock the localStorage
+global.window.localStorage = storageMock();
+// mock the sessionStorage
+global.window.sessionStorage = storageMock();
```

Comment:
Don't we want to mock local & session storage with functions: mockLocalStorage() & mockSessionStorage()? And add the function a-la `restore`, doing `global.window.localStorage = nativeLocalStorage`?

Reply:
Sure ! :)